java/azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/PartitionReceiver.java

```diff
@@ -291,17 +292,32 @@ public final long getEpoch()
 			}
 		});
 	}
+	
+	/**
+	 * Register a receive handler that will be called when an event is available. A 
+	 * {@link PartitionReceiveHandler} is a handler that allows user to specify a callback
+	 * for event processing and error handling in a receive pump model. 
+	 * 
+	 * @param receiveHandler An implementation of {@link PartitionReceiveHandler}. Setting this handler to <code>null</code> will stop the receive pump.
+	 */
+	public void setReceiveHandler(final PartitionReceiveHandler receiveHandler)
+	{
+		this.setReceiveHandler(receiveHandler, false);
+	}
 
 	/**
 	 * Register a receive handler that will be called when an event is available. A 
 	 * {@link PartitionReceiveHandler} is a handler that allows user to specify a callback
 	 * for event processing and error handling in a receive pump model. 
 	 * @param receiveHandler An implementation of {@link PartitionReceiveHandler}
+	 * @param invokeOnTimeout flag to indicate whether the {@link PartitionReceiveHandler#onReceive(Iterable)} should be invoked when the receive call times out
 	 */
-	public void setReceiveHandler(final PartitionReceiveHandler receiveHandler)
+	public void setReceiveHandler(final PartitionReceiveHandler receiveHandler, final boolean invokeOnTimeout)
```

Comment:
>invokeOnTimeout [](start = 91, length = 15)

wording of the argument is implying that we can get TimeoutException which is not really correct. I would suggest we say something more like "invokeOnNullEvents" to be more explicit.